onnxruntime/python/ort_trainer.py

```diff
@@ -275,15 +275,20 @@ def convert_model_loss_fn_to_onnx(model, loss_fn, model_desc, device):
     input_names = [input.name_ for input in model_desc.inputs_]
     output_names = [output.name_ for output in model_desc.outputs_]
 
-    sample_inputs = []
-    for input_desc in model_desc.inputs_:
-        input_sample = generate_sample(input_desc, device)
-        sample_inputs.append(input_sample)
-
-    sample_outputs = []
-    for output_desc in model_desc.outputs_:
-        output_sample = generate_sample(output_desc, device)
-        sample_outputs.append(output_sample)
+    if isinstance(inputs, torch.Tensor):
+        inputs = [inputs]
+    if isinstance(inputs, dict):
+        sample_inputs = [inputs[k.name_].to(device=device) for k in model_desc.inputs_]
+    else:
```

Comment:
instead of `else` to handle lists implicitly, what do you think about using `elif instance(inputs, list)` for lists and an `else` to raise exception due to invalid format for `inputs`? We could catch bad input faster by making sure we get what we expect